mmpose/datasets/datasets/mesh/mesh_base_dataset.py

```diff
@@ -0,0 +1,156 @@
+        self.image_info = {}
+        self.ann_info = {}
+
+        self.annotations_path = ann_file
+        self.img_prefix = img_prefix
+        self.pipeline = pipeline
+        self.test_mode = test_mode
+
+        self.ann_info['image_size'] = np.array(data_cfg['image_size'])
+        self.ann_info['iuv_size'] = np.array(data_cfg['iuv_size'])
+        self.ann_info['num_joints'] = data_cfg['num_joints']
+        self.ann_info['flip_pairs'] = None
+        self.db = []
+        self.pipeline = Compose(self.pipeline)
+
+        # flip_pairs
+        # For all mesh dataset, we use 24 joints as CMR and SPIN.
+        self.ann_info['flip_pairs'] = [[0, 5], [1, 4], [2, 3], [6, 11],
+                                       [7, 10], [8, 9], [20, 21], [22, 23]]
+        self.ann_info['use_different_joint_weights'] = False
+        self.ann_info['joint_weights'] = \
+            np.ones(24, dtype=np.float32).reshape(
+                (self.ann_info['num_joints'], 1))
+
+        self.ann_info['uv_type'] = data_cfg['uv_type']
+        self.ann_info['use_IUV'] = data_cfg['use_IUV']
+        self.iuv_prefix = os.path.join(
+            self.img_prefix, '{}_IUV_gt'.format(self.ann_info['uv_type']))
+        self.db = self._get_db(ann_file)
+
+    def _get_db(self, ann_file):
+        """Load dataset."""
+        data = np.load(ann_file)
+        tmpl = dict(
+            image_file=None,
+            center=None,
+            scale=None,
+            rotation=0,
+            joints_2d=None,
+            joints_2d_visible=None,
+            joints_3d=None,
+            joints_3d_visible=None,
+            gender=None,
+            pose=None,
+            beta=None,
+            has_smpl=0,
+            iuv_file=None,
+            has_iuv=0)
+        gt_db = []
+
+        _imgnames = data['imgname']
+        _scales = data['scale'].astype(np.float32)
+        _centers = data['center'].astype(np.float32)
+        dataset_len = len(_imgnames)
+
+        # Get 2D keypoints
+        if 'part' in data.keys():
+            _keypoints = data['part'].astype(np.float32)
+        else:
+            _keypoints = np.zeros((dataset_len, 24, 3), dtype=np.float32)
+
+        # Get gt 3D joints, if available
+        if 'S' in data.keys():
+            _joints_3d = data['S'].astype(np.float32)
+        else:
+            _joints_3d = np.zeros((dataset_len, 24, 4), dtype=np.float32)
+
+        # Get gt SMPL parameters, if available
+        if 'pose' in data.keys() and 'shape' in data.keys():
+            _poses = data['pose'].astype(np.float32)
+            _betas = data['shape'].astype(np.float32)
+            has_smpl = 1
+        else:
+            _poses = np.zeros((dataset_len, 72), dtype=np.float32)
+            _betas = np.zeros((dataset_len, 10), dtype=np.float32)
+            has_smpl = 0
+
+        # Get gender data, if available
+        if 'gender' in data.keys():
+            _genders = data['gender']
+            _genders = np.array([0 if str(g) == 'm' else 1
```

Comment:
```
_genders = np.array([str(g) != 'm' for g in _genders]).astype(np.int32)
```

Reply:
btw why gender has the type of int32?

Reply:
In some datasets, there is no gender label, and we use `-1` to represent the unknown gender. So gender is not a bool, and I choose int32 for simplicity.

Reply:
what about np.int vs np.int32?

Reply:
It's ok, I will change to np.int.